src/interceptors/ClientRequest/NodeClientRequest.ts

```diff
@@ -398,9 +423,8 @@ export class NodeClientRequest extends ClientRequest {
     encoding?: BufferEncoding | null,
     callback?: ClientRequestEndCallback | null
   ): this {
-    // Set the response source to "bypass".
-    // Any errors emitted past this point are not suppressed.
-    this.responseSource = 'bypass'
+    this.state = HttpClientInternalState.ResponseReceived
+    this.responseType = 'passthrough'
```

Comment:
I kept the `responseSource`, just renamed it to `responseType`. The `ResponseReceived` phase alone doesn't imply the response type, and that's correct. That way request phases can be filtered out like I've shown in the first comment. But the `responseType` is an additional information that's important in certain scenarios, such as error suppression.